buildkite/resource_test_suite.go

```diff
@@ -78,10 +105,26 @@ func (ts *testSuiteResource) Create(ctx context.Context, req resource.CreateRequ
 	payload["show_api_token"] = true
 	payload["team_ids"] = []string{teamOwnerUuid}
 
+	// Construct URL to call to the REST API
 	url := fmt.Sprintf("/v2/analytics/organizations/%s/suites", ts.client.organization)
-	err = ts.client.makeRequest(ctx, "POST", url, payload, &response)
-	if err != nil {
-		resp.Diagnostics.AddError("Failed to create test suite", err.Error())
+	createErr := retry.RetryContext(ctx, timeout, func() *retry.RetryError {
+		err = ts.client.makeRequest(ctx, "POST", url, payload, &response)
+
+		if err != nil {
+			if isRetryableError(err) {
+				return retry.RetryableError(err)
+			}
+			return retry.NonRetryableError(err)
+		}
+
+		return nil
+	})
```

Comment:
I wonder if the whole thing should be a single retry? Maybe not, I don't really know but calling it out if anyone has an argument either way

Reply:
Fair point - I _think_ it makes sense leaving them seperate, as the first technically is a read, the second is the actual creation. Latest commit has the first using the `Read` timeout, and the second (REST POST) using the `Create` timeout via the provider

Reply:
Fair point - I think it makes sense leaving them seperate, as the first technically is a read, the second is the actual creation. Latest commit has the first using the Read timeout, and the second (REST POST) using the Create timeout via the provider